EngUpdater/GitHub.cs

```diff
@@ -212,6 +214,126 @@ public static async Task<(string sha, DateTimeOffset date) []> GetCommits(string
                 yield return PullRequest.Parse(je);
             }
         }
+
+        public static void Trace (string message, bool verbose)
+        {
+            if (verbose)
+                Console.WriteLine (message);
+        }
+
+        /// <summary>
+        /// Reset and clean current working directory. This will ensure that the current
+        /// working directory matches the current Head commit.
+        /// </summary>
+        /// <param name="repo">Repository whose current working directory should be operated on.</param>
+        public static void ResetAndCleanWorkingDirectory(IRepository repo)
+        {
+            // Reset the index and the working tree.
+            repo.Reset(ResetMode.Hard);
+
+            // Clean the working directory.
+            repo.RemoveUntrackedFiles();
+        }
+        public static void PrepareMonoWorkingDirectory (Repository repo,
```

Comment:
`PrepareMonoWorkingDirectory`, and `CleanupUnusedBranches` doesn't really need to be mono specific. I have local changes that fix that up though, so your changes here are fine.